src/calories.rs

```diff
@@ -0,0 +1,92 @@
+#[derive(Debug, PartialEq)]
+pub struct CalorieCounter {
+    third_max: usize,
+    second_max: usize,
+    pub max: usize,
+    current_counter: usize,
+}
+
+impl CalorieCounter {
+    pub fn new(
+        third_max: usize,
+        second_max: usize,
+        max: usize,
+        current_counter: usize,
+    ) -> CalorieCounter {
+        CalorieCounter {
+            third_max,
+            second_max,
+            max,
+            current_counter,
+        }
+    }
+
+    pub fn reset() -> CalorieCounter {
+        CalorieCounter::new(0, 0, 0, 0)
```

Comment:
You could just derive `Default` on `CalorieCounter` and then use `Default::default()` here instead.